src/Microsoft.VisualStudio.ProjectSystem.CSharp/Microsoft.VisualStudio.ProjectSystem.CSharp.csproj

```diff
@@ -35,10 +35,27 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="GlobalSuppressions.cs" />
+    <Compile Include="ProjectSystem\Rules\CSharp.CscCommandLineArgs.cs">
+      <DependentUpon>CSharp.CscCommandLineArgs.xaml</DependentUpon>
+    </Compile>
+    <Compile Include="ProjectSystem\Rules\CSharpProjectProperties.cs" />
     <Compile Include="ProjectSystem\SpecialFileProviders\CSharpAssemblyInfoSpecialFileProvider.cs" />
   </ItemGroup>
   <ItemGroup>
     <None Include="project.json" />
   </ItemGroup>
+  <ItemGroup>
+    <XamlPropertyRule Include="ProjectSystem\Rules\CSharp.CscCommandLineArgs.xaml">
+      <Generator>MSBuild:GenerateRuleSourceFromXaml</Generator>
+      <SubType>Designer</SubType>
+    </XamlPropertyRule>
+  </ItemGroup>
   <Import Project="..\..\build\Targets\VSL.Imports.targets" />
+  <ItemDefinitionGroup>
+    <!-- Change the defaults for all rules so that they generate to C# properties -->
```

Comment:
So even the properties in the BrowseObject rules are in CSharpProjectProperties? Shouldn't it be just the ones published to the Project context?